org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/codeactions/CodeActionMarkerResolution.java

```diff
@@ -81,12 +81,10 @@ public void run(IMarker marker) {
 				}
 			}
 			if (wrapper != null) {
-				if (codeAction.getEdit() == null) {
-					if (CodeActionCompletionProposal.isCodeActionResolveSupported(wrapper.getServerCapabilities())) {
-						CodeAction resolvedCodeAction = wrapper.execute(ls -> ls.getTextDocumentService().resolveCodeAction(codeAction)).get(2, TimeUnit.SECONDS);
-						if (resolvedCodeAction != null) {
-							codeAction = resolvedCodeAction;
-						}
+				if (CodeActionCompletionProposal.isCodeActionResolveSupported(wrapper.getServerCapabilities())) {
+					CodeAction resolvedCodeAction = wrapper.execute(ls -> ls.getTextDocumentService().resolveCodeAction(codeAction)).get(2, TimeUnit.SECONDS);
+					if (resolvedCodeAction != null) {
+						codeAction = resolvedCodeAction;
 					}
 				}
 				if (codeAction.getEdit() != null) {
```

Comment:
Because of the check if (codeAction.getEdit() != null) at the beginning of the method with an early exit. This condition here can never be true meaning this whole block is redundant